openff/toolkit/utils/exceptions.py

```diff
@@ -71,45 +68,218 @@ class LicenseError(ToolkitUnavailableException):
 
 
-class InvalidIUPACNameError(MessageException):
+class InvalidIUPACNameError(OpenFFToolkitException):
     """Failed to parse IUPAC name"""
 
 
-class AntechamberNotFoundError(MessageException):
+class AntechamberNotFoundError(OpenFFToolkitException):
     """The antechamber executable was not found"""
 
 
-class ParseError(MessageException, ValueError):
+class SMILESParseError(ValueError):
     """The record couple not be parsed into the given format"""
+
+
+class InvalidConformerError(Exception):
+    """
+    This error is raised when the conformer added to the molecule
+    has a different connectivity to that already defined.
+    or anyother conformer related issues.
+    """
+
+
+class SmilesParsingError(Exception):
+    """
+    This error is rasied when parsing a smiles string results in an error.
+    """
+
+
+class NotAttachedToMoleculeError(OpenFFToolkitException):
+    """Exception for when a component does not belong to a Molecule object, but is queried"""
+
+
+class DuplicateUniqueMoleculeError(OpenFFToolkitException):
+    """
+    Exception for when the user provides indistinguishable unique molecules when trying to identify atoms from a PDB
+    """
+
+
+class NotBondedError(OpenFFToolkitException):
+    """
+    Exception for when a function requires a bond between two atoms, but none is present
+    """
+
+
+class InvalidBoxVectorsError(OpenFFToolkitException):
+    """
+    Exception for setting invalid box vectors
+    """
+
+
+class InvalidPeriodicityError(OpenFFToolkitException):
+    """
+    Exception for setting invalid periodicity
+    """
+
+
+class MissingUniqueMoleculesError(OpenFFToolkitException):
+    """
+    Exception for a when unique_molecules is required but not found
+    """
+
+
+class SMIRKSMismatchError(OpenFFToolkitException):
+    """
+    Exception for cases where smirks are inappropriate
+    for the environment type they are being parsed into
+    """
+
+
+class SMIRKSParsingError(OpenFFToolkitException):
+    """
+    Exception for when SMIRKS are not parseable for any environment
+    """
+
+
+class ParameterHandlerRegistrationError(OpenFFToolkitException):
+    """
+    Exception for errors in ParameterHandler registration
+    """
+
+
+class SMIRNOFFVersionError(OpenFFToolkitException):
+    """
+    Exception thrown when an incompatible SMIRNOFF version data structure in attempted to be read.
+    """
+
+
+class SMIRNOFFAromaticityError(OpenFFToolkitException):
+    """
+    Exception thrown when an incompatible SMIRNOFF aromaticity model is checked for compatibility.
+    """
+
+
+class _DeprecatedParseError(Exception):
+    """DEPRECATED: Error for when a SMIRNOFF data structure is not parseable by a ForceField"""
+
+    def __init__(self, msg):
+        super().__init__(msg)
+        self.msg = msg
+
+    def __str__(self):
+        return self.msg
+
+
+class SMIRNOFFParseError(OpenFFToolkitException, _DeprecatedParseError):
```

Comment:
Could I suggest sticking to only one of XXParseError or XXParsingError so it's easier to remember? So far I've seen:

* SMILESParseError
* SmilesParsingError
* SMIRKSParsingError
* SMIRNOFFParseError

Reply:
Totally - the SMILES duplicates were my mistake, and I have no memory of interacting with the SMIRKS error so it must be purely copy-pasted. Any suggestions which one?  I suppose `XXParsingError` rolls off the tongue better but I don't really care.

Reply:
Looking at the existing exceptions there's not a lot of Verb/ing ones to follow. Although ParsingError sounds nicer, I might lean the other way though and prefer XXParseError because that's what RDKit writes out when it has a parsing error.